### PR TITLE
#31 - Add code coverage support improve

### LINKED ISSUE #31: Improve code-coverage & add GH check action
## Background
Improve or add code coverage support to be able measure code quality.

## Feature
New ability to measure current code coverage as one of QA metrics.
Add GH action to check changed file coverage.

### PR DESCRIPTION
**Pull request by miroslavpojer**
- Used Absa fork of JaCoCo solution - with scala method filtering.
- Added support for code coverage GitHub action check as part of PR.

Closes #31

### CODE REVIEW

**Review by Zejnilovic — COMMENTED** [review]
All good. We will need to add classes with `@Configuration` to the excludes. 

- za.co.absa.logingw.rest.Application
- za.co.absa.logingw.rest.SecurityConfig
- za.co.absa.logingw.rest.WebMvcConfig

You will never be able to properly unit test these. These should be covered by Integration tests.

@miroslavpojer can I ask you to add them in this PR?

**Comment by miroslavpojer** [comment]
> All good. We will need to add classes with `@Configuration` to the excludes.
> 
> * za.co.absa.logingw.rest.Application
> * za.co.absa.logingw.rest.SecurityConfig
> * za.co.absa.logingw.rest.WebMvcConfig
> 
> You will never be able to properly unit test these. These should be covered by Integration tests.
> 
> @miroslavpojer can I ask you to add them in this PR?

Sure, I will prepare excludes for them.

File: .github/workflows/jacoco_check.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#
+# Copyright 2022 ABSA Group Limited
```
Comment (Zejnilovic) [inline .github/workflows/jacoco_check.yml:2]:
```suggestion
# Copyright 2023 ABSA Group Limited
```

**Review by Zejnilovic — COMMENTED** [review]
Interesting. Would this mean JaCoCo knows of Spring classes? Or is that just "luck"? I will look more into this.

**Comment by miroslavpojer** [comment]
> Interesting. Would this mean JaCoCo knows of Spring classes? Or is that just "luck"? I will look more into this.

JaCoCo can be a little unpredictable and most of expectations goes from real world practice.
Basic answer is - if the class can be detected and instrumented by JaCoCo then it can be measured.
I hope I will learn more about this process when I will trying to add method filtration feature into origin project.

**Review by Zejnilovic — APPROVED** [review]

**Review by dk1844 — APPROVED** [review]

**Pull request merged**